panel/pane/base.py

```diff
@@ -386,6 +386,7 @@ def _update_object(
                         parent.children[node] = new_models  # type: ignore
                         break
             elif isinstance(parent, _BkTabs):
+                parent.tabs = cast(list[_BkTabPanel], parent.tabs)
```

Comment:
@mattpap, This shouldn't be needed, in my opinion. This conversion always happens on the Bokeh site of things:

https://github.com/bokeh/bokeh/blob/4693080b6763222cf37680f62518944855ef4540/src/bokeh/models/layouts.py#L586-L589 

Maybe [default factory functions](https://docs.python.org/3/library/dataclasses.html#default-factory-functions) could be a way to reduce this overhead.

Reply:
Currently for simplicity `.accepts()` clauses where merged into property types in `*.pyi` files. In the future, relevant declarations will have be split into setters and getters with asymmetric types, if possible at all.